addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/alerts/AlertManager.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.websocket.alerts;
+
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+
+public class AlertManager {
+
+    private ExtensionAlert extensionAlert;
+
+    public AlertManager(ExtensionAlert extensionAlert) {
+        this.extensionAlert = extensionAlert;
+    }
+
+    public void alertFound(WebSocketAlertWrapper alertWrapper) {
+        extensionAlert.alertFound(alertWrapper.getAlert(), alertWrapper.getHandshakeReference());
```

Comment:
The `extensionAlert` will be null when the extension is disabled. We should add a null check here or change the `hook` to not set up the passive scan.